codice/ddf
# [2.21.x] DDF-5572 Render map marker labels with custom text notation on 2D and 3D maps

#### What does this PR do?
Backport of #5638 and #5768 

Adds an option and input field to the Search UI preferences to enable "custom text notation". Custom text notation allows for a user to enter a metacard attribute to display as a text label next to metacard map markers in Intrigue. title is the default attribute used.

#### Who is reviewing it? 

@aaronilovici 
@josephthweatt 
@mdang8 

#### Select relevant component teams: 

@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@vinamartin 
@andrewkfiedler 

#### How should this be tested?

1. Open the Admin console and navigate to `Search UI -> Configuration -> Catalog UI Search`
2. Scroll down to `Show Custom Text Notations` and click the checkbox to enable it
3. Ingest some metacards in Intrigue, ensuring there are metacards that overlap with different titles
4. Open the 2D map and run a * search
5. Verify the metacard map markers have a label next to each one and that there are no overlapping labels
6. Scroll in and out to change appearances of labels
7. Select different overlapping metacards and verify that only the selected metacard's label shows up (no overlapping labels)
8. Turn on clustering and verify that clusters do not have labels
9. Turn off clustering and verify that unclustered metacards do not have overlapping labels
10. Re-run the search and verify the labels still display as intended
11. Repeat steps 4-10 in the 3D map

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #5572, #5508, #5767 

#### Screenshots

![image](https://user-images.githubusercontent.com/6109231/73222200-a9ba0e80-4130-11ea-80b8-00440412a758.png)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
